src/plugins/postgres_permissions_operator.py

```diff
@@ -0,0 +1,171 @@
+        self.schema_url = schema_url
+        self.db_schema = db_schema
+        self.profiles = profiles
+        self.role = role
+        self.scope = scope
+        self.dry_run = dry_run
+        self.create_roles = create_roles
+        self.revoke = revoke
+
+    def execute(self, context=None):
+        """Executes the 'apply_schema_and_profile_permissions' method from schema-tools to set the database permissions on objects to roles"""
+
+        # setup logger so output can be added to the Airflow logs
+        logger = logging.getLogger(__name__)
+
+        # setup database connection where the database objects are present
+        engine = _get_engine(self.db_conn)
+
+        # Option ONE: batch grant
+        if self.batch_ind:
+
+            # get current datetime and make it aware of the TZ (mandatory)
+            now = pendulum.now("Europe/Amsterdam")
+
+            # calculate the delta between current datetime and specified time window
+            time_window_hour = int(self.batch_timewindow.split(':')[0])
+            time_window_minutes = int(self.batch_timewindow.split(':')[1])
+            delta = now - timedelta(hours=time_window_hour, minutes=time_window_minutes)
+
+            logger.info(f"the time window is set starting at: {delta} till now")
+
+            # setup an Airflow session to access Airflow repository data
+            session = Session()
+
+            # get list of dags that meet time window and state outcome
+            # it uses the Airflow DagRun class to get data
+            executed_dags_after_delta = [
+                dag.dag_id
+                for dag in session.query(DagRun)
+                .filter(DagRun.execution_date > delta)
+                .filter(DagRun._state == "success")
+                # exclude the dag itself that calls this batch grant method
+                .filter((DagRun.dag_id != "airflow_db_permissions"))
+            ]
+
+
+            if executed_dags_after_delta:
+
+                for dataset_name in executed_dags_after_delta:
+
+                    logger.info(
+                            f"set grants for {dataset_name}"
+                        )
+
+                    try:
+                        ams_schema = schema_defs_from_url(
+                            schemas_url=self.schema_url,
+                            dataset_name=dataset_name,
+                        )
+
+                        apply_schema_and_profile_permissions(
+                            engine=engine,
+                            pg_schema=self.db_schema,
+                            ams_schema=ams_schema,
+                            profiles=self.profiles,
+                            role=self.role,
+                            scope=self.scope,
+                            dry_run=self.dry_run,
+                            create_roles=self.create_roles,
+                            revoke=self.revoke,
+                        )
+
+                    except:
```

Comment:
Een bare except, dat is eigenlijk niet de bedoeling in python. Het verbaast met dat deze door flake8 heen is gekomen. Het is altijd de bedoeling om zo specifiek mogelijk te zijn qua exceptions.

Reply:
Helemaal gelijk, pas ik aan